# A Single Cell data analysis for Chron’s Disease with COTAN
2023 Computational Health Laboratory project - MD Computer Science University of Pisa.

## Group Members
- [Andrea Piras](https://github.com/aprs3)
- [Matteo Ninniri](https://github.com/Asduffo)

## Project description
This project makes single-cell RNA sequencing analysis with COTAN and more: In details it performs data cleaning, clustering of the cells, gene set enrichment. Given multiple patients analyzed can output for each cell the best intersection of genes between the patients. This application also can find the
exclusive genes for each condition (healthy, inflamed, not-inflamed samples) dividing them by new genes found in the enrichment and known genes from a given reference. Last, for each patient can perform the pathway enrichment of the genes found previeusly.

## How to run
Clone the repository
```
git clone https://github.com/aprs3/CHL_Project.git
```
Install COTAN and his dependencies in R
```
devtools::install_github("seriph78/COTAN")
```

Download the [dataset](https://singlecell.broadinstitute.org/single_cell/study/SCP1884/human-cd-atlas-study-between-colon-and-terminal-ileum#study-download) files in the correct dataset folder. Compress with gzip the scp.barcodes.tsv, scp.features.tsv, scp.raw.mtx. Then rename them features.tsv.gz, barcodes.tsv.gz, matrix.mtx.gz.

## Scripts and helper files description
This is the list of R scripts with the instruction of how to use them:
* **main.R**: Performs the data cleaning, clustering, and gene set enrichment analysis using COTAN. Outputs
multiple cluster cuts from a given range as .csv files. It can be run by either setting the arguments by command
line, or by setting the args array manually such that the first and second element of the list are respectively
the dataset folder name and the patient’s ID. The script will create all the folders automatically. Also, the
script will calculate the clusterization for the targeted enrichment genes for all the clusters ranging between
two variable, start and end. If one of the args is set to -1, the script will procedurally ask the user to insert
variables values in an interactive way.
* **venn.R**: Given a list of patients, it calculates the optimal number of enriched genes clusters for each patient
(within a range set by the variables start and end) such that the clusters where a cell type’s markers are placed
variates as least as possible. Finally, it saves into a .txt file the optimal number of clusters for each patient
and the .csv file with the intersections found with the method described in section 5.1. To execute it, set the
variable dataset with the selected dataset name, the list to load with the patients IDs and the variables start,
end that specify the range where to search the optimal number of clusters.
* **find state markers.R**: Using the intersection files generated by venn.R, this script outputs the genes which
behave constantly like certain cell markers, for each patient condition (healthy, inflamed, and non-inflamed
intestine) with the method described in section 5.2. To execute, set the string dataset with the selected
dataset’s name and three lists N (not inflamed), H (healthy), I (inflamed) with the patients’ codes.
1
* **separate state markers.R**: This script splits the genes found with find state markers.R into genes present in
known cells genes.csv and not present in such list. To execute this script, follow the same instructions specified
in find state markers.R.
* **utils.R**: Various utilities for reading files, as well as displaying the various plots

List of helper files:
* **known cells genes.csv**: a copy of this file is present in each dataset. They contain the set of cells found in the
datasets by Kong et al. [dataset](https://singlecell.broadinstitute.org/single_cell/study/SCP1884/human-cd-atlas-study-between-colon-and-terminal-ileum#study-download), as well as their gene markers.
* **enrichment list.txt**: the list of genes we were requested to perform gene enrichment analysis on (contained in enrichment_list.txt).

## Folders structure and files description
The folder structure is the same for each dataset, so for simplicity reasons we describe as example "TI_IMM" with one patient and their files.

```
TI_IMM
├── dataset
│   ├── features.scp.gz
│   ├── barcodes.scp.gz
│   ├── matrix.mtx.gz
└── H101694
    ├── enrichment_csv
    │   ├── TI_IMM_H101694_EnrichmentGenesClusters_x.csv: Enrichment genes clusters with "x" number of cuts
    └── plot
        ├── TI_IMM_H101694__00_ECDPlot.pdf
        ├── TI_IMM_H101694__01_CellSizePlot.pdf: original library size plot
        ├── TI_IMM_H101694__02_GenesSizePlot.pdf: original gene size plot
        ├── TI_IMM_H101694__03_MitocondrialPlot.pdf: original mitochondrial cells plot
        ├── TI_IMM_H101694__04_CellSizePlot_cut_n.pdf: library size plot after n number of cuts
        ├── TI_IMM_H101694__05_GeneCountPlot_cut_n.pdf: gene size plot after n number of cuts
        ├── TI_IMM_H101694__06_MitocondrialCount_cut_n.pdf: mitochondrial cells plot after n number of cuts
        ├── TI_IMM_H101694__07_PCACells.pdf: A and B clusters
        ├── TI_IMM_H101694__08_PCACellsBRemoval.pdf: PCA after B cluster removal (if removed)
        ├── TI_IMM_H101694__09_CleanPlotGenes.pdf
        ├── TI_IMM_H101694__10_CleanPlotUDE.pdf: UDE correlation plot
        ├── TI_IMM_H101694__11_CleanPlotNu.pdf: Nu plot for cells efficiency visualization
        ├── TI_IMM_H101694__12_PCACells_cut_1.pdf
        ├── TI_IMM_H101694__13_CleanPlotGenes_cut_1.pdf
        ├── TI_IMM_H101694__14_CleanPlotUDE_cut_1.pdf
        ├── TI_IMM_H101694__15_CleanPlotNu_cut_1.pdf
        ├── TI_IMM_H101694__16_GDIPlot.pdf: GDI plot
        ├── TI_IMM_H101694__22_FineClustersSummary.pdf: Summary of the clusters statistics
        ├── TI_IMM_H101694__23_MergedClustersSummary.pdf: Summary of the merged clusters statistics (not realiable)
        ├── TI_IMM_H101694__26_clustersMarkersHeatmapPlot.pdf: Markers heatmap with visual clustering colors
        ├── TI_IMM_H101694__27_enrichmentHm.pdf: Enrichment heatmap clustered but not cutted
        ├── TI_IMM_H101694__27_enrichmentHm_X.pdf: Enrichment heatmap wih x number of cuts
        └── TI_IMM_H101694__28_enrichmentHmUnclustered.pdf: Enrichment heatmap without clustering
```



## Resources
- [COTAN](https://github.com/seriph78/COTAN/tree/devel)
- [Human CD atlas study between colon and terminal ileum](https://singlecell.broadinstitute.org/single_cell/study/SCP1884/human-cd-atlas-study-between-colon-and-terminal-ileum#study-summary)

